lib/blacklight/request_builders.rb

```diff
@@ -51,6 +51,14 @@ def solr_search_params_logic= logic
       deprecation_deprecate :solr_search_params_logic=
     end
 
+    def search_builder_class
```

Comment:
Agree, I've wanted to do what jcoyne wants to do too. But allowing the defaults in configuration might make the upgrade path smoother from current BL -- there are a lot of Blacklight plugins (gems) that are going to break and need to be redone, unless there's a good way to keep the backwards-compat api too.